components/lsp/LSP.tsx

```diff
@@ -95,45 +95,29 @@ const Testing = () => {
       contract.getCurrentTime().then((ts: ethers.BigNumber) => {
         setCurrentTime(ts.toString());
       });
+      const erc20 = createERC20ContractInstance(signer, data.collateralToken);
 
-      contract.collateralToken().then(async (res: any) => {
-        const erc20 = createERC20ContractInstance(signer, res);
-        erc20.decimals().then((decimals: ethers.BigNumber) => {
-          setCollateralDecimals(decimals.toString());
-        });
-
-        const balance = (await erc20.balanceOf(account)) as ethers.BigNumber;
+      erc20.balanceOf(account).then((balance: ethers.BigNumber) => {
         setCollateralBalance(balance);
-        setERC20Contract(erc20);
-      });
-
-      contract.collateralPerPair().then((cpp: ethers.BigNumber) => {
-        setCollateralPerPair(cpp);
-      });
-
-      contract.longToken().then((addr: string) => {
-        setLongTokenAddress(addr);
       });
 
-      contract.shortToken().then((addr: string) => {
-        setShortTokenAddress(addr);
-      });
+      setERC20Contract(erc20);
```

Comment:
Same here, why not call this something with collateral in the name

Reply:
renamed, I'll change the props down as well.